src/api/authAPI.ts

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+interface FormValue {
+  id: string;
+  password: string;
+}
+
+export const loginAPI = async (
+  id: string,
+  password: string,
+): Promise<FormValue> => {
+  try {
+    const res = await axios.post('/login', {
+      id: id,
+      password: password,
+    });
+    return res.data;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const joinAPI = async (id: string, password: string): Promise<void> => {
+  try {
+    await axios.post('/join', {
+      id: id,
+      password: password,
+    });
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const getUserAPI = async (): Promise<FormValue> => {
+  try {
+    const res = await axios.get('/user');
```

Comment:
```suggestion
    const {data} = await axios.get<AxiosResponse<FormResponse>>('/user');
    return data.data;
```

이렇게 하면 hook에서도 타입추론될꺼에요

Reply:
<img width="552" alt="스크린샷 2022-10-28 오전 1 03 09" src="https://user-images.githubusercontent.com/20167165/198341166-e9fcbbc9-1e9f-4879-871d-1f6f5f7a2d2d.png">